packages/debug/src/browser/view/debug-threads-widget.ts

```diff
@@ -23,10 +24,13 @@ import { injectable, inject } from "inversify";
 export class DebugThreadsWidget extends VirtualWidget {
     private _threads: DebugProtocol.Thread[] = [];
     private _threadId?: number;
+    private _threadStatus: Map<number, boolean> = new Map();
```

Comment:
Get rid of `_threadStatus` field. Rely on `debugSession.debugSessionState`

Reply:
My only problems is that inside of the onContinuedEvent it assumes that to remove the thread you need to have the event but the debug adapter protocol suggests that the continued event is optional after a continue:

```java
 /** Event message for 'continued' event type.
        The event indicates that the execution of the debuggee has continued.
        Please note: a debug adapter is not expected to send this event in response to a request that implies that execution continues, e.g. 'launch' or 'continue'.
        It is only necessary to send a ContinuedEvent if there was no previous request that implied this.
    */```

Because if the onContinuedEvent state is not triggered then we won't accurately know the state of all the threads

Reply:
I see.
Let's emit `continued` event ourselves after continue response. In this case `DebugSessionState` will be consistent.

```
             const event: DebugProtocol.ContinuedEvent = {
                    type: 'event',
                    seq: 1,
                    event: 'continued',
                    body: {
                        threadId,
                        allThreadsContinued: false
                    }
                };
                this.proceedEvent(event);
```